e2e_tests/tests/task/test_generic_tasks.py

```diff
@@ -0,0 +1,281 @@
+    while t < timeout and not found:
+        resp = bindings.get_GetTask(test_session, taskId=task_id)
+        found = expected_state == resp.task.taskState
+        t += 1
+        time.sleep(1)
+    return found
+
+
+@pytest.mark.e2e_cpu
+def test_create_generic_task() -> None:
+    """
+    Start a simple task with a context directory called from the task CLI
+    """
+    command = [
+        "det",
+        "-m",
+        conf.make_master_url(),
+        "task",
+        "create",
+        conf.fixtures_path("generic_task/test_config.yaml"),
+        "--context",
+        conf.fixtures_path("generic_task"),
+    ]
+
+    subprocess.run(command, universal_newlines=True, stdout=subprocess.PIPE, check=True)
+
+
+@pytest.mark.e2e_cpu
+def test_generic_task_completion() -> None:
+    """
+    Start a simple task and check for task completion
+    """
+    test_session = api_utils.determined_test_session()
+
+    with open(conf.fixtures_path("generic_task/test_config.yaml"), "r") as config_file:
+        # Create task
+        config_text = config_file.read()
+        context_directory = context.read_v1_context(
+            pathlib.Path(conf.fixtures_path("generic_task")), []
+        )
+
+        req = bindings.v1CreateGenericTaskRequest(
+            config=config_text,
+            contextDirectory=context_directory,
+            projectId=None,
+            forkedFrom=None,
+            parentId=None,
+            inheritContext=False,
+            noPause=False,
+        )
+        task_resp = bindings.post_CreateGenericTask(test_session, body=req)
+
+        # Check for complete state
+        timeout = 30
+        is_valid_state = wait_for_task_state(
+            test_session, task_resp.taskId, bindings.v1GenericTaskState.COMPLETED, timeout
+        )
+        if not is_valid_state:
+            pytest.fail(f"task failed to complete after {timeout} seconds")
+
+
+@pytest.mark.e2e_cpu
+def test_create_generic_task_error() -> None:
+    """
+    Start a simple task that fails and check for error task state
+    """
+    test_session = api_utils.determined_test_session()
+
+    with open(conf.fixtures_path("generic_task/test_config_error.yaml"), "r") as config_file:
+        # Create task
+        config_text = config_file.read()
+        context_directory = context.read_v1_context(
+            pathlib.Path(conf.fixtures_path("generic_task")), []
+        )
+
+        req = bindings.v1CreateGenericTaskRequest(
+            config=config_text,
+            contextDirectory=context_directory,
+            projectId=None,
+            forkedFrom=None,
+            parentId=None,
+            inheritContext=False,
+            noPause=False,
+        )
+        task_resp = bindings.post_CreateGenericTask(test_session, body=req)
+
+        # Check for error state
+        timeout = 30
+        is_valid_state = wait_for_task_state(
+            test_session, task_resp.taskId, bindings.v1GenericTaskState.ERROR, timeout
+        )
```

Comment:
I like that you are labeling the `30` as `timeout`, which is great for readability, but there is a better way:

```python
is_valid_state = wait_for_task_state(
    test_session, task_resp.taskId, bindings.v1GenericTaskState.ERROR, timeout=30
)
```

This is:
- more idiomatic python
- fewer lines of code
- easier to read